pkg/asset/manifests/proxy.go

```diff
@@ -0,0 +1,139 @@
+type Proxy struct {
+	FileList []*asset.File
+}
+
+var _ asset.WritableAsset = (*Proxy)(nil)
+
+// Name returns a human-friendly name for the asset.
+func (*Proxy) Name() string {
+	return "Proxy Config"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate
+// the asset.
+func (*Proxy) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+		&Networking{},
+	}
+}
+
+// Generate generates the Proxy config and its CRD.
+func (p *Proxy) Generate(dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	network := &Networking{}
+	dependencies.Get(installConfig, network)
+
+	config := &configv1.Proxy{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: configv1.SchemeGroupVersion.String(),
+			Kind:       "Proxy",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+			// not namespaced
+		},
+	}
+
+	if installConfig.Config.Proxy != nil {
+		config.Spec = configv1.ProxySpec{
+			HTTPProxy:  installConfig.Config.Proxy.HTTPProxy,
+			HTTPSProxy: installConfig.Config.Proxy.HTTPSProxy,
+			NoProxy:    installConfig.Config.Proxy.NoProxy,
+		}
+	}
+
+	if config.Spec.HTTPProxy != "" || config.Spec.HTTPSProxy != "" {
+		config.Status = configv1.ProxyStatus{
+			HTTPProxy:  installConfig.Config.Proxy.HTTPProxy,
+			HTTPSProxy: installConfig.Config.Proxy.HTTPSProxy,
+			NoProxy:    CreateNoProxy(installConfig),
+		}
+	}
+
+	configData, err := yaml.Marshal(config)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create %s manifests from InstallConfig", p.Name())
+	}
+
+	p.FileList = []*asset.File{
+		{
+			Filename: proxyCfgFilename,
+			Data:     configData,
+		},
+	}
+
+	return nil
+}
+
+// CreateNoProxy adds platform-specific NO_PROXY values to the user-provided values
+// Platform values are: serviceCIDR, podCIDR, localhost, 127.0.0.1, api.clusterdomain,
+// api-int.clusterdomain, etcd-idx.clusterdomain
+func CreateNoProxy(installConfig *installconfig.InstallConfig) string {
+	defaults := []string{
+		"127.0.0.1",
+		"localhost",
+	}
+
+	defaults = append(defaults, installConfig.Config.Networking.ServiceNetwork[0].String())
+	defaults = append(defaults, fmt.Sprintf("api.%s", installConfig.Config.ClusterDomain()))
+	defaults = append(defaults, fmt.Sprintf("api-int.%s", installConfig.Config.ClusterDomain()))
+
+	for i := int64(0); i < *installConfig.Config.ControlPlane.Replicas; i++ {
+		etcdHost := fmt.Sprintf("etcd-%d.%s", i, installConfig.Config.ClusterDomain())
+		defaults = append(defaults, etcdHost)
+	}
+
+	for _, v := range installConfig.Config.Networking.ClusterNetwork {
+		defaults = append(defaults, v.CIDR.String())
+	}
+
+	return CombineValues(installConfig.Config.Proxy.NoProxy, defaults)
+}
+
+// CombineValues filters any duplicates between user- and platform-provided values to create
+// a comma-separated list of unique no-proxy values
+func CombineValues(user string, platform []string) string {
```

Comment:
this function is pretty generic and not related to NO_PROXY, can we reword to use no_proxy as example...

also make this function private